packages/klighd-core/src/proxy-view/proxy-view.tsx

```diff
@@ -217,66 +230,92 @@ export class ProxyView extends AbstractUIExtension {
 
         //// Stacking order ////
-        const orderedOffScreenNodes = this.orderNodes(opacityOffScreenNodes, canvas);
+        const orderedOffScreenNodes = this.orderNodes(opacityOffScreenNodes, canvasLRF);
 
         //// Use proxy-rendering as specified by synthesis ////
         const synthesisRenderedOffScreenNodes = this.getSynthesisProxyRendering(orderedOffScreenNodes, ctx);
 
         //// Calculate transformations ////
-        const size = Math.min(canvas.width, canvas.height) * this.sizePercentage;
+        const size = Math.min(canvasCRF.width, canvasCRF.height) * this.sizePercentage;
         const transformedOffScreenNodes = synthesisRenderedOffScreenNodes.map(({ node, proxyBounds }) => ({
             node,
-            transform: this.getTransform(node, size, proxyBounds, canvas)
+            transform: this.getTransform(node, size, proxyBounds, canvasCRF)
         }));
+        const fromPercent = 0.01;
+        const offsetCRF = Math.min(canvasCRF.width, canvasCRF.height) * fromPercent;
+        const offsetLRF = Math.min(canvasLRF.width, canvasLRF.height) * fromPercent;
 
         //// Apply clustering ////
-        const clusteredNodes = this.applyClustering(transformedOffScreenNodes, size, canvas);
+        const clusteredNodes = this.applyClustering(transformedOffScreenNodes, size, canvasCRF);
 
-        //// Create edges ////
-        const edges = this.routeEdges(clusteredNodes, onScreenNodes, canvas, ctx);
+        //// Route edges to proxies ////
+        const routedEdges = this.routeEdges(clusteredNodes, onScreenNodes, canvasCRF, offsetCRF, ctx);
+
+        //// Connect off-screen edges ////
+        const connectors = this.connectEdges(root, canvasLRF, offsetLRF, ctx);
 
         //// Render the proxies ////
         const proxies = [];
         this.currProxies = [];
 
-        // Start with edges since they should never appear in front of nodes
-        for (const edge of edges) {
-            // Create an edge proxy
-            const edgeProxy = this.createEdgeProxy(edge, ctx);
-            if (edgeProxy) {
-                proxies.push(edgeProxy);
-            }
-        }
-
         // Nodes
         for (const { node, transform } of clusteredNodes) {
             // Create a proxy
-            const proxy = this.createProxy(node, transform, canvas, ctx);
+            const proxy = this.createProxy(node, transform, canvasLRF, ctx);
             if (proxy) {
                 proxies.push(proxy);
                 this.currProxies.push({ proxy, transform });
             }
         }
 
+        const edgeProxies = [];
+        // Edges
+        const edges = ([] as { edge: SKEdge, transform: TransformAttributes }[]).concat(
+            connectors.overlayEdges, // Start with overlays to not have overlays over proxy edges
+            connectors.proxyEdges,
+            routedEdges.overlayEdges, // But routing overlays should still be over connectors
```

Comment:
I think they're convenient as you know exactly what part of the code the comment refers to. In this case, I would otherwise need to say which edges are routing overlays. What do you think of
```
const edges = ([] as { edge: SKEdge, transform: TransformAttributes }[]).concat(
    // Start with overlays to not have overlays over proxy edges
    connectors.overlayEdges,
    connectors.proxyEdges,
    // But routing overlays should still be over connectors
    routedEdges.overlayEdges, 
    ...
```
as a middle ground?

Reply:
my issue is that I think this muddles readability a bit, because it makes these look like distinct lines of code which they aren't, but maybe it's not really an issue. I'm open for other opinions on the matter